angular2-jwt.ts

```diff
@@ -109,7 +109,19 @@ export class AuthHttp {
       this.setGlobalHeaders(this.config.globalHeaders, providedOpts);
     }
 
-    newOptions = newOptions.merge(new RequestOptions(providedOpts));
+    let providedOptions = new RequestOptions(providedOpts);
+    if (newOptions.headers && newOptions.headers.keys().length > 0) {
+      newOptions.headers.forEach((header: Object, name: string) => {
+
+        // set only not existing header keys
+        if (!providedOptions.headers.has(name)) {
+          let headerValue: string = (header as any)[Object.keys(header)[0]];
+          providedOptions.headers.set(name, headerValue);
+        }
+      });
+    }
+
+    newOptions = newOptions.merge(providedOptions);
```

Comment:
This whole block seems really awkward.

for every header in newOptions, set that header in providedOptions if it isn't already set, then merge provided options into new options.

Which headers is kept?